### ISSUE TITLE
strptime and to_datetime documentation is lacking

### ISSUE THREAD

**Issue by huxi** [pola-rs/polars#11580]
### Description

The documentation of [strptime](https://docs.rs/polars/latest/polars/prelude/string/struct.StringNameSpace.html#method.strptime) is unclear. [to_datetime](https://docs.rs/polars/latest/polars/prelude/string/struct.StringNameSpace.html#method.to_datetime) has the same issue.

I don't understand what `ambiguous` (added in 0.33, I think) is doing, even after diving into the source down to `map_many_private`.

Kind of lost, I simply ended up making the compiler happy by adding `false.into()` as `ambiguous` argument.
It "works" but I have no idea what I'm doing.

### Link

https://docs.rs/polars/latest/polars/prelude/string/struct.StringNameSpace.html#method.strptime

**Comment by deanm0000** [comment]
The Python docs have more info https://pola-rs.github.io/polars/py-polars/html/reference/expressions/api/polars.Expr.str.strptime.html#polars.Expr.str.strptime

**Comment by MarcoGorelli** [comment]
I'd suggest setting 'raise' as the default if you don't want to use it

You're right about docs though, thanks for the issue

**Comment by huxi** [comment]
How would I create a `raise` `Expr` in Rust? The `Expr` type for that parameter is part of my problem in understanding the API. Shouldn't this be an `enum` instead?

The Python docs are indeed sufficient to understand the parameter. Will keep in mind to check them in the future, thanks.

**Comment by MarcoGorelli** [comment]
does `lit("raise")` work?

**Comment by huxi** [comment]
I think I'm lost again regarding understanding what's going on.

From the Python docs, I thought I understood it the following way:
If you have a pattern like `"%Y-%m-%d %H:%M"` then `earliest` would mean that seconds and milliseconds of the datetime would be set to 0 and `latest` would set them to 59 for seconds and 999 for milliseconds.

But it seems to be related to time zones and DST instead.

I just found this in `to_date` so `lit("raise")` seems to be correct.

```rust
self.strptime(DataType::Date, options, lit("raise"))
```

**Comment by MarcoGorelli** [comment]
> But it seems to be related to time zones and DST instead.

Yup that's right! Suggestions on how to clarify would be welcome

**Comment by rben01** [comment]
On the Rust side, shouldn't this be an enum? `HandleAmbiguity::Raise` or something similar. As far as I can tell, the `Expr` is just used to get the `StringChunked` out of it, and then only the first element of that `StringChunked` is ever examined.

**Comment by MarcoGorelli** [comment]
it can be an entire column (you can have a different value for each element)
